src/test/java/org/jenkinsci/plugins/workflow/cps/DSLTest.java

```diff
@@ -314,6 +314,17 @@ public void namedSoleParamForStep() throws Exception {
         p.setDefinition(new CpsFlowDefinition("try {def c = classLoad(getClass().name); error(/did not expect to be able to load ${c} from ${c.classLoader}/)} catch (ClassNotFoundException x) {echo(/good, got ${x}/)}", false));
         r.assertBuildStatusSuccess(p.scheduleBuild2(0));
     }
+
+    /**
+     *  Test user defined closure execution
+     */
+    @Test public void userDefinedClosureExecution() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("binding.setVariable(\"my_closure\", { echo \"my closure!\" })\n my_closure() ", false));
```

Comment:
Two things that are weak about this test:

* It does not verify that `CpsClosure2.call(Object)` properly handles arguments, which is not at all obvious to me at least from code inspection—Groovy is weird this way. Ought to be testing closures defining zero, one, multiple, and varargs just to be sure.
* It does not verify that `CpsCallableInvocation` is being handled naturally. This would be most easily tested by, say,
  ```groovy
  binding['my_closure'] = {
    sleep 1 // exercises CPS interpreter
    echo 'ran'
  }
  my_closure()
  ```

Reply:
Thanks for the feedback @jglick, i'll modify the test case.